File: app/src/biology/MetabolicSynthesis.js
export class MetabolicSynthesis {
  /**
   * Administra la química y dualidad de recursos:
   * Biomasa = Combustible basal (Energía pura de supervivencia)
   * Mineral = Material constitutivo (Para reproducirse y construir Estigmergia)
   */
  constructor(initialBiomass = 100, initialMineral = 0) {
    this.biomass = initialBiomass;
    this.mineral = initialMineral;

    this.maxBiomass = 200;
    this.maxMineral = 100;
  }

  decay(baseDrain, environmentalHeat) {
    // Heat = -1 to 1. 
    // Calor extra aumenta el gasto metabólico (se mueven más rápido). 
    // Frío lo aletarga (hibernación relativa).
    let heatModifier = 1.0 + (environmentalHeat * 0.5); 
    this.biomass -= (baseDrain * heatModifier);
  }

  ingest(type, amount) {
    if (type === 'BIOMASS') {
        this.biomass = Math.min(this.maxBiomass, this.biomass + amount);
    } else if (type === 'MINERAL') {
        this.mineral = Math.min(this.maxMineral, this.mineral + amount);
    }
  }

  canReproduce() {
    // La reproducción ahora es biológica (requiere biomasa) y sintética (requiere sílice/mineral)
    return this.biomass > 100 && this.mineral > 30;
  }

  consumeForReproduction() {
    this.biomass -= 60;
    this.mineral -= 30; // Costo genético y estructural
  }

  consumeForStigmergy(cost) {
    if (this.mineral >= cost) {
        this.mineral -= cost;
        return true;
    }
    return false;
  }

  isDead() {
    return this.biomass <= 0;
  }
}

File: GodotEngine/src/biology/MetabolicSynthesis.cs
using Godot;

public class MetabolicSynthesis
{
    public float Biomass;
    public float Mineral;
    public float MaxBiomass = 200f;
    public float MaxMineral = 100f;
    
    // Feature: Fertilidad dinámica controlada desde Main UI
    public static float FertilityThreshold = 100f;

    public MetabolicSynthesis(float initialBiomass = 100f, float initialMineral = 0f)
    {
        Biomass = initialBiomass;
        Mineral = initialMineral;
    }

    public void Decay(float baseDrain, float environmentalHeat)
    {
        // Calor de -1 a 1
        float heatModifier = 1.0f + (environmentalHeat * 0.5f);
        Biomass -= (baseDrain * heatModifier);
    }

    public void Ingest(string type, float amount)
    {
        if (type == "BIOMASS")
            Biomass = Mathf.Min(MaxBiomass, Biomass + amount);
        else if (type == "MINERAL")
            Mineral = Mathf.Min(MaxMineral, Mineral + amount);
    }

    public bool CanReproduce() => Biomass > FertilityThreshold && Mineral > 30f;

    public void ConsumeForReproduction()
    {
        Biomass -= (FertilityThreshold * 0.6f); // Gasta el 60% del umbral exigido
        Mineral -= 30f;
    }

    public bool ConsumeForStigmergy(float cost)
    {
        if (Mineral >= cost) {
            Mineral -= cost;
            return true;
        }
        return false;
    }
    
    public bool IsDead() => Biomass <= 0f || float.IsNaN(Biomass);
}
